include/openPMD/RecordComponent.hpp

```diff
@@ -174,7 +175,15 @@ RecordComponent::storeChunk(Offset o, Extent e, std::shared_ptr<T> data)
         throw std::runtime_error("Unallocated pointer passed during chunk store.");
     Datatype dtype = determineDatatype(data);
     if( dtype != getDatatype() )
-        throw std::runtime_error("Datatypes of chunk and dataset do not match.");
+    {
+        std::ostringstream oss;
+        oss << "Datatypes of chunk ("
+            << getDatatype()
```

Comment:
@C0nsultant did I switch those? is `dtype` of the incoming data the current `chunk` or the `dataset`? ;-)

Reply:
yep, I think I mixed it up. will also rename `of chunk` to `of chunk data`

Reply:
[```getDatatype()```](https://github.com/openPMD/openPMD-api/blob/0.4.0-alpha/src/backend/BaseRecordComponent.cpp#L47-L51) is equal to the datatype assigned to a dataset/```RecordComponent``` through [```resetDataset()```](https://github.com/openPMD/openPMD-api/blob/0.4.0-alpha/src/RecordComponent.cpp#L46-L60).
So yes, it is actually the wrong way around.

Reply:
thx, already fixed locally. will push the rebase when #324 is in :)

Reply:
updated :)